src/NovaVault.sol

```diff
@@ -1,21 +1,30 @@
 // SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.8.13;
+pragma solidity 0.8.17;
 
-import {Errors} from "./libraries/Errors.sol";
 import {INovaVault} from "./interfaces/INovaVault.sol";
 import {INovaAdapterBase} from "./interfaces/INovaAdapterBase.sol";
-import {IERC20} from "./interfaces/IERC20.sol";
+import {ERC20} from "@solmate/tokens/ERC20.sol";
+import {SafeTransferLib} from "@solmate/utils/SafeTransferLib.sol";
+import {ReentrancyGuard} from "@solmate/utils/ReentrancyGuard.sol";
+import {Errors} from "./libraries/Errors.sol";
+
+contract NovaVault is INovaVault, ReentrancyGuard {
+    using SafeTransferLib for ERC20;
 
-contract NovaVault is INovaVault {
     mapping(address => address) public _novaAdapters;
     address immutable sDAI;
-    event Referral(uint16 referral, address indexed depositor, uint256 amount);
 
+    modifier isAddressZero(address asset) {
```

Comment:
bad naming IMO. `isAddressZero` looks like a function that returns a boolean.
you can name this modifier `onlyNonZero` for instance